Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm

```diff
@@ -740,6 +769,248 @@ static void checkAuthorizationOptions(bool userActionInitiated, String initiator
+
+    HashMap<String, String> redirectHeaders;
+    redirectHeaders.add("location"_s, "http://127.0.0.1:8091/simple.html"_s);
+    TestWebKitAPI::HTTPResponse redirectResponse(301, WTFMove(redirectHeaders));
+
+    server.addResponse("/redirection.html"_s, WTFMove(redirectResponse));
+
+    navigationCompleted = false;
+
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500)]);
+    // A separate delegate that implements decidePolicyForNavigationAction.
+    auto delegate = adoptNS([[TestSOAuthorizationDelegate alloc] init]);
+    configureSOAuthorizationWebView(webView.get(), delegate.get(), OpenExternalSchemesPolicy::Allow);
+    [delegate setIsDefaultPolicy:false];
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://127.0.0.1:8091/redirection.html"]]];
+    Util::run(&authorizationPerformed);
+    checkAuthorizationOptions(false, "null"_s, 0);
+    EXPECT_TRUE(policyForAppSSOPerformed);
+
+    auto response = adoptNS([[NSHTTPURLResponse alloc] initWithURL:[NSURL URLWithString:@"http://127.0.0.1:8091/simple.html"] statusCode:302 HTTPVersion:@"HTTP/1.1" headerFields:@{ @"Location" : @"http://127.0.0.1:8091/simple2.html" }]);
+    [gDelegate authorization:gAuthorization didCompleteWithHTTPResponse:response.get() httpBody:adoptNS([[NSData alloc] init]).get()];
+    Util::run(&navigationCompleted);
+    navigationCompleted = false;
+
+#if PLATFORM(IOS_FAMILY)
+    Util::run(&navigationCompleted);
+    navigationCompleted = false;
+#endif
+
+    EXPECT_WK_STREQ(finalURL, "http://127.0.0.1:8091/simple2.html");
+}
+
+TEST(SOAuthorizationRedirect, InterceptionSucceedWith307Simple)
+{
+    resetState();
+    SWIZZLE_SOAUTH(PAL::getSOAuthorizationClass());
+
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500)]);
+    auto delegate = adoptNS([[TestSOAuthorizationDelegate alloc] init]);
+    configureSOAuthorizationWebView(webView.get(), delegate.get(), OpenExternalSchemesPolicy::Allow);
+
+    TestWebKitAPI::HTTPServer server(TestWebKitAPI::HTTPServer::UseCoroutines::Yes, [&] (Connection connection) -> Task {
+        while (true) {
+            auto request = co_await connection.awaitableReceiveHTTPRequest();
+            auto path = HTTPServer::parsePath(request);
+            if (path == "/simple2.html"_s) {
+                request.append(0);
+                EXPECT_TRUE(strnstr(request.data(), "POST /simple2.html HTTP/1.1\r\n", request.size()));
+                EXPECT_TRUE(strnstr(request.data(), "\r\nContent-Type: application/x-www-form-urlencoded\r\n", request.size()));
+                EXPECT_TRUE(strnstr(request.data(), "\r\n\r\nname=test", request.size()));
+            }
+            co_await connection.awaitableSend(HTTPResponse(SimpleHtml).serialize());
+        }
+    }, TestWebKitAPI::HTTPServer::Protocol::Http, 8091);
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[server.request() URL]]];
```

Comment:
This is redundant.  Just use server.request instead of extracting the URL and putting it back into a request.

Reply:
Done